polkadot/runtime/parachains/src/paras_inherent/tests.rs

```diff
@@ -666,12 +666,6 @@ mod enter {
 			// * 3 disputes.
 			assert_eq!(limit_inherent_data.disputes.len(), 2);
 
-			assert_ok!(Pallet::<Test>::enter(
-				frame_system::RawOrigin::None.into(),
-				limit_inherent_data,
-			));
-
-			// TODO [now]: this assertion fails with async backing runtime.
```

Comment:
Why is it fine to remove this assert?

Reply:
My understanding is that each `ParachainsInherentData` is intended to be filtered by `create_inherent_inner` and then entered using `enter`. 

But both of these call `process_inherent_data` which makes a bunch of state changes within the inclusion and scheduler pallets regardless of calling context. Those state changes were prematurely draining assignments from the mock assigner, causing the test to fail.

I'm not clear on why or whether we need to call `create_inherent` and `enter` from the same runtime instance. But if that is desirable then `process_inherent_data` needs to be modified and the test assertion can be left as is.